src/main/java/nl/smallproject/www/techiteasy/controllers/WallBracketsController.java

```diff
@@ -0,0 +1,61 @@
+package nl.smallproject.www.techiteasy.controllers;
+
+import jakarta.validation.Valid;
+import nl.smallproject.www.techiteasy.dtos.WallBracket.WallBracketInputDto;
+import nl.smallproject.www.techiteasy.dtos.WallBracket.WallBracketOutputDto;
+import nl.smallproject.www.techiteasy.dtos.WallBracket.WallBracketUpdateDto;
+import nl.smallproject.www.techiteasy.models.WallBracket;
+import nl.smallproject.www.techiteasy.services.WallBracketService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/wallbracket")
+public class WallBracketsController {
+    private final WallBracketService wallBracketService;
+
+    public WallBracketsController(WallBracketService wallBracketService) {
+        this.wallBracketService = wallBracketService;
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<List<WallBracketOutputDto>> getAllWallBrackets() {
+        List<WallBracketOutputDto> allWallBrackets = wallBracketService.getAllWallBrackets();
+        return ResponseEntity.ok(allWallBrackets);
+    }
+
+    @RequestMapping(value = "{id}", method = RequestMethod.GET)
+    public ResponseEntity<WallBracketOutputDto> getWallBracketById(@PathVariable Long id) {
+        WallBracketOutputDto wallBracketOutputDto = wallBracketService.getWallBracketById(id);
+        return ResponseEntity.ok(wallBracketOutputDto);
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    public ResponseEntity<Object> createWallBracket(@Valid @RequestBody WallBracketInputDto wallBracketInputDto) {
+        WallBracket newWallBracket = wallBracketService.createWallBracket(wallBracketInputDto);
+
+        URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest()
+                .path("/" + newWallBracket.getId())
+                .buildAndExpand(newWallBracket)
+                .toUri();
+
+        return ResponseEntity.created(location).build();
+    }
+
+    @RequestMapping(value = "{id}", method = RequestMethod.PUT)
+    public ResponseEntity<Object> updateWallBracket(@PathVariable Long id, @Valid @RequestBody WallBracketUpdateDto wallBracketUpdateDto) {
```

Comment:
Persoonlijk zou ik hier een 'InputDTO' van maken ipv 'UpdateDTO', maar dat is denk ik een kwestie van smaak

Reply:
Het is prima om een specifieke dto te maken, maar er lijkt nu geen verschil te zijn tussen InputDto en UpdateDto.